### PR TITLE
feat: duplicate team and user collections

### PR DESCRIPTION
**Pull request by balub**
Closes HSB-425

### What's changed

In this PR we are adding a new mutation to duplicate team and user collections.

```
For TeamCollections,

duplicateTeamCollection(collectionID: string) : Boolean

For UserCollections,

duplicateUserCollection(collectionID: string, reqType: ReqType) : Boolean
```

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Additional Information

none

### CODE REVIEW

File: packages/hoppscotch-backend/src/user-collection/user-collection.resolver.ts [hunk 1]
```diff
@@ -390,6 +390,35 @@ export class UserCollectionResolver {
...
+  async duplicateUserCollection(
+    @GqlUser() user: AuthUser,
+    @Args({
+      name: 'collectionID',
+      description: 'ID of the collection',
+    })
+    collectionID: string,
+    @Args({
+      name: 'reqType',
+      description: 'Type of UserCollection',
```
Comment (jamesgeorge007) [inline packages/hoppscotch-backend/src/user-collection/user-collection.resolver.ts:406]:
```suggestion
      description: 'Type of UserCollection',
      type: () => ReqType
```

File: docker-compose.yml [hunk 2]
```diff
@@ -112,7 +112,7 @@ services:
     build:
       dockerfile: packages/hoppscotch-backend/Dockerfile
       context: .
-      target: prod
+      target: dev
```
Comment (AndrewBastin) [inline docker-compose.yml:115]: Is this change intentional ?

Comment (balub) [reply docker-compose.yml:115]: Yup, it is done during dev. After the PR Is reviewed I revert it back to prod before merge.

**Review by AndrewBastin — APPROVED** [review]

File: packages/hoppscotch-backend/src/team-collection/team-collection.service.ts [hunk 3]
```diff
@@ -1446,4 +1446,35 @@ export class TeamCollectionService {
...
+      collection.right.teamID,
+      collectionID,
+    );
+    if (E.isLeft(collectionJSONObject)) return E.left(TEAM_INVALID_COLL_ID);
+
+    const result = await this.importCollectionsFromJSON(
+      JSON.stringify([
+        {
+          ...collectionJSONObject.right,
+          name: `${collection.right.title} - duplicate`,
```
Comment (jamesgeorge007) [inline packages/hoppscotch-backend/src/team-collection/team-collection.service.ts:1470]:
```suggestion
          name: `${collection.right.title} - Duplicate`,
```

File: packages/hoppscotch-backend/src/user-collection/user-collection.service.ts [hunk 4]
```diff
@@ -1138,4 +1138,45 @@ export class UserCollectionService {
...
+      collectionID,
+    );
+    if (E.isLeft(collectionJSONObject))
+      return E.left(collectionJSONObject.left);
+
+    const result = await this.importCollectionsFromJSON(
+      JSON.stringify([
+        {
+          ...collectionJSONObject.right,
+          name: `${collection.right.title} - duplicate`,
```
Comment (jamesgeorge007) [inline packages/hoppscotch-backend/src/user-collection/user-collection.service.ts:1171]:
```suggestion
          name: `${collection.right.title} - Duplicate`,
```
